src/gamemode.c

```diff
@@ -535,6 +535,21 @@ handle_unregister_game_by_pidfd (XdpDbusGameMode *object,
   return G_DBUS_METHOD_INVOCATION_HANDLED;
 }
 
+static void
+client_properties_changed (GDBusProxy* proxy,
+                           GVariant* changed_properties,
+                           char** invalidated_properties)
+{
+  g_autoptr(GVariant) count = NULL;
+  gboolean enabled = FALSE;
+
+  count = g_dbus_proxy_get_cached_property (proxy, "ClientCount");
```

Comment:
You should probably check if any one of the invalidated properties is `ClientCount` before updating the D-Bus property. Something like `if (!g_strv_contains (invalidated_properties, "ClientCount")) \n return;`

Reply:
Ah, that's why I added [G_DBUS_PROXY_FLAGS_GET_INVALIDATED_PROPERTIES](https://docs.gtk.org/gio/flags.DBusProxyFlags.html#get_invalidated_properties) -- because the invalidated properties dance is annoying if you really do need the value.

But I could change that and query for the value when it's invalidated if you don't like using the flag.

Reply:
I could add a bit of optimization here and check if ClientCount is actually in changed properties. But I was going for simplicity over 100% correctness. But that might not be the style move here.

Reply:
It's not a massive deal, but in my experience we have to be careful not to update properties when they didn't actually change, as this could lead to all sorts of odd behaviors on apps.